### ISSUE TITLE
PDF Button fails with

### ISSUE THREAD

**Issue by fstorch** [joomlahenk/fabrik#580]
**To Reproduce**
Started to try PDF's with Beta2c. Still fails in Beta3
Only occurs in Migrated F3 Site
Activate PDF Button in a Form which will generate this link:
http://site/index.php?option=com_fabrik&view=details&formid=5&rowid=817&format=pdf&lang=de

**Expected behavior**
Clicking will generate and open a PDF of the form.

**Actual behavior**
Error: _There is no "template.cassiopeia.ltr" asset of a "preset" type in the registry._ 

**Additional context**
Fails also with default template bootstrap
Does not fail with a new Fabrik 4 site.

**Comment by trob** [comment]
I can't replicate. 
Is it working in the backend?
Do you have Cassiopeia as J! template?
Is this a multilanguage site? 

Different URL: It seems on your new F4 you have SEF URLs enabled

**Comment by fstorch** [comment]
Not working in the backend, but I hoenstly never tested the forms (or list's) in backend before.
So both - F3 and F4 - result in
_View not found [name, type, prefix]: details, pdf, fabrikadminView_ 
Either when trying to use the pdf link or when just opening an entry from the list.

Yes, I'm using Cassiopeia.
Yes, this is multilanguage. (tried both languages)

No, SEF URL's are not enabled.

Fabrik cache is refreshed.
Global Cache is cleared

**Comment by trob** [comment]
I still can't replicate even with a multi language site.

> http://site/index.php/component/fabrik/details/2/1?format=pdf

This URL IS with SEF enabled.
Do you have the original Fabrik bootstrap templates without layout overrides?
Or custom ones with custom PDF links?
Or forms in modules?

**Comment by fstorch** [comment]
Correction.
PDF is actually working in backend.
Just need to use the correct View/Edit butoon to open the form.
This PDF Link works:
http://site/administrator/index.php?option=com_fabrik&task=details.view&format=pdf&formid=2&rowid=809
It's looking terrible, but I expected a lot of .CSS work required.

Above URL was taken from the Fresh F4 site and yes, SEF was active there. (Sorry, I thought I checked that)
It is still not on the prod site, so the issue is not URL related.

Set all templates to bootstrap but error _There is no "template.cassiopeia.ltr" asset of a "preset" type in the registry._ 
remains in Frontend.
No forms in modules.

**Comment by fstorch** [comment]
Not sure what could be the problem.
![grafik](https://user-images.githubusercontent.com/34415681/205459095-e53348a7-bc50-4ea2-8d4b-fc54082d6266.png)

**Comment by trob** [comment]
No idea.
Do you have Cassiopeia as frontend template?
Can you enable Joomla's System debug and see if you get an error stack?

**Comment by fstorch** [comment]
Yes, using Cassiopeia.

And yes, here's the callstack:
There is no "template.cassiopeia.ltr" asset of a "preset" type in the registry.                    
                                            <div>
                            

Call stack
--
# | Function | Location
1 | () | JROOT\libraries\src\WebAsset\WebAssetRegistry.php:134
2 | Joomla\CMS\WebAsset\WebAssetRegistry->get() | JROOT\libraries\src\WebAsset\WebAssetManager.php:263
3 | Joomla\CMS\WebAsset\WebAssetManager->useAsset() | JROOT\libraries\src\WebAsset\WebAssetManager.php:197
4 | Joomla\CMS\WebAsset\WebAssetManager->__call() | JROOT\templates\cassiopeia\component.php:50
5 | require() | JROOT\libraries\src\Document\HtmlDocument.php:736
6 | Joomla\CMS\Document\HtmlDocument->_loadTemplate() | JROOT\libraries\src\Document\HtmlDocument.php:791
7 | Joomla\CMS\Document\HtmlDocument->_fetchTemplate() | JROOT\libraries\src\Document\HtmlDocument.php:608
8 | Joomla\CMS\Document\HtmlDocument->parse() | JROOT\libraries\src\Application\CMSApplication.php:1004
9 | Joomla\CMS\Application\CMSApplication->render() | JROOT\libraries\src\Application\SiteApplication.php:714
10 | Joomla\CMS\Application\SiteApplication->render() | JROOT\libraries\src\Application\CMSApplication.php:299
11 | Joomla\CMS\Application\CMSApplication->execute() | JROOT\includes\app.php:61
12 | require_once() | JROOT\index.php:32

</div>

**Comment by fstorch** [comment]
Trying to debug and found (clicking on the PDF Button) in
C:/wamp64/www/conacare/components/com_fabrik/models/form.php.FabrikFEModelForm->getFormCss(): lineno 659	
In Backend, the $tmplPath is set to 
`"templates/atum/html/com_fabrik/details/bootstrap/template_css.php?c=2&amp;rowid=809&amp;view=details&amp;rowid=809"	
`
atum exists in Administrator though no com_fabrik:
![grafik](https://user-images.githubusercontent.com/34415681/205502500-3549f9b4-0b83-4e55-b48b-a297e42df8c6.png)
Neither here:
![grafik](https://user-images.githubusercontent.com/34415681/205502532-8ffd7d4a-feaa-4d4d-8bcd-003fff5b2863.png)
Nevertheless the PDF opens.

In Frontend:
$tmplPath	"templates/cassiopeia_conacare/html/com_fabrik/details/bootstrap/template_css.php?c=2&amp;rowid=809&amp;view=details&amp;rowid=809"	
cassiopeia_conacare is a child of Cassiopeia Template:
![grafik](https://user-images.githubusercontent.com/34415681/205502843-26e8e477-2ed5-41cc-aac0-3a39fc2bf77f.png)

Before I only had this file:
c:\wamp64\www\conacare\templates\cassiopeia_conacare\templateDetails.xml
I now copied the remaining files from Cassiopeia (though I assume it is not required)
But the problem persist.
site template has no com_fabrik either:
![grafik](https://user-images.githubusercontent.com/34415681/205503126-5ea003d7-bac6-47f5-a379-54e262386c64.png)
![grafik](https://user-images.githubusercontent.com/34415681/205503198-eef35beb-36ff-45d1-84d3-12ef7710c83b.png)
![grafik](https://user-images.githubusercontent.com/34415681/205503280-3b9b1090-2bc6-4932-9ef6-6a858ee6fbeb.png)

c:\wamp64\www\conacare\templates\cassiopeia_conacare\joomla.asset.json exists and has
```
   {
      "name": "template.cassiopeia.ltr",
      "type": "preset",
      "dependencies": [
        "template.cassiopeia.ltr#style",
        "template.cassiopeia#script"
      ]
    },
```
nothing in media/...

**Comment by trob** [comment]
Ok.
So you have

- not the original Cassiopeia template but a child template?
- not the original Fabrik bootstrap template but an override inside your Cassiopeia child template?
Pretty non-standard...

Is this templates/cassiopeia_conacare/html/com_fabrik/details/bootstrap/ an original Fabrik4 bootstrap copy?

Can you try with the original Fabrik details/bootstrap template? (rename  templates/cassiopeia_conacare/html/com_fabrik/details/bootstrap to something else)

 It's always better to create a custom details (PDF) template in components/com_fabrik/views/details/tmpl and select this as PDF template.

**Comment by trob** [comment]
I tested your setup (created a Cassiopeia child and also an Atum child):

It seems Fabrik PDF is not compatible with child templates at the moment.

**Comment by fstorch** [comment]
Switched back to Cassiopeia Parent as Default Template and the PDF Button worked fine (except the ugly output, which will keep me busy during the next week(s) 😊)
Sorry to have waisted your time, but the root cause wasn't clear to me.
When I played with user.css I have switched back to the Parent already, but at that time found no difference so I used the child again.
I also tested with Fabrik bootstrap template, but as we know now, that doesn't make a difference.
I have used a custom template forpdf output in site\components\com_fabrik\views\details\tmpl\pdf_conacare\
since years.

**Comment by trob** [comment]
Yes, I assume this "$tmplPath "templates/cassiopeia_conacare/html/com_fabrik/details/bootstrap/template_css.php?c=2&rowid=809&view=details&rowid=809" is not coming from a real template located there but is part of this issue.

But at the moment this "child template issue" is a corner case (no idea if coming from Fabrik or from J! itself), so low prio.

**Comment by fstorch** [comment]
Agree. Can be easily avoided and shouldn't be a showstopper.
